src/OpenTelemetry.AspNetCore/OpenTelemetryServicesExtensions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// <copyright file="OpenTelemetryServicesExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+using OpenTelemetry.Instrumentation.AspNetCore;
+using OpenTelemetry.Instrumentation.Dependencies;
+using OpenTelemetry.Trace.Configuration;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up OpenTelemetry services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class OpenTelemetryServicesExtensions
+    {
+        /// <summary>
+        /// Adds OpenTelemetry services to the specified <see cref="IServiceCollection" /> with automatic data collection.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="openTelemetryOptions">The <see cref="OpenTelemetryOptions" /> to configure.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddOpenTelemetryWithInstrumentation(
+            this IServiceCollection services,
+            OpenTelemetryOptions openTelemetryOptions = null)
+        {
+            services.AddOpenTelemetry(builder =>
+            {
+                if (openTelemetryOptions == null || openTelemetryOptions.EnableDependencyInstrumentation)
```

Comment:
This pattern has the disadvantage of depending on the defaults to be `true`, using the default instance if the parameter was `null` makes the code clear and straightforward to change the defaults in a single location.

Reply:
Just to be clear here I'm saying to use `openTelemetryOptions ??= default;` and remove `openTelemetryOptions == null` from the if statements

Reply:
Hi @pjanotti , just changed!